src/cli/serve/deprecated_config.js

```diff
@@ -0,0 +1,16 @@
+import { forOwn, has, noop } from 'lodash';
+
+// deprecated settings are still allowed, but will be removed at a later time. They
+// are checked for after the config object is prepared and known, so legacySettings
+// will have already been transformed.
+export const deprecatedSettings = new Map([
+  [['server', 'xsrf', 'token'], 'server.xsrf.token is deprecated. It is no longer used when providing xsrf protection.']
```

Comment:
Given how this data will be used, I think a simple hash would be better here.

1. The inputs are private and won't be changing at runtime.
2. You never need to do things like access the "size" of the collection.
3. The key/value pairs are entered by humans and are likely to be the thing changing most frequently in this file over time.

So basically, the `Map` here seems to be optimizing for CPU time rather than for humans, which I don't think is the right priority in this file.

Of course, objects can't have arrays as keys, but that brings me to my next point: I think these keys should be strings instead of arrays.  Again, optimizing for humans.

I sort of think about these key/values as configuration data rather than programmatic values.  For the sake of fewer moving parts and a simpler and less-bug-prone implementation, I don't suggest that we pull them out into an actual configuration file or anything, but treating them as if they could have come from a configuration file seems appropriate.